Projects/App/Sources/Screens/ProfileScreen/Screens/DeleteAccount/Slack/SlackBot.swift

```diff
@@ -0,0 +1,124 @@
+import Foundation
+import hGraphQL
+import Apollo
+import hCore
+import Flow
+
+class SlackBot {
+    @Inject var client: ApolloClient
+    
+    enum SlackError: Error {
+        case invalidRequestBody(description: String)
+        case requestError(description: String)
+        case invalidStatusCode
+        case emptyDataReceived
+        case badResponse
+    }
+    
+    private let url: URL = URL(string: "https://slack.com/api/chat.postMessage")!
+    
+    func postSlackMessage(
+        memberDetails: MemberDetails
+    ) -> Future<Bool> {
+        client.fetch(
+            query: GraphQL.SlackDetailsQuery()
```

Comment:
First it fetches the slack app's `oAuthToken` and the channels `channelId` from Giraffe through this query.